friends/FriendsHandler.js

```diff
@@ -0,0 +1,243 @@
+      body: JSON.stringify({ stack: err.stack, message: err.message })
+    }));
+};
+
+/*
+ * GET /friends/request/
+ */
+module.exports.getRequests = async (event, context) => {
+  context.callbackWaitsForEmptyEventLoop = false;
+
+  const token = event.headers.Authorization;
+  const decoded = decodeJwt(token);
+  if (!decoded) // token empty or invalid
+    return { 
+      statusCode: 403,
+      body: JSON.stringify({ message: 'Forbidden: missing or invalid JWT.'  })
+    };
+  
+  return await getRequests(decoded.id)
+    .then(resp => ({
+      statusCode: 200,
+      body: JSON.stringify(resp)
+    }))
+    .catch(err => ({
+      statusCode: err.statusCode || 500,
+      body: JSON.stringify({ stack: err.stack, message: err.message })
+    }));
+};
+
+/*
+ * POST /friends/request/send/
+ */
+module.exports.sendRequest = async (event, context) => {
+  context.callbackWaitsForEmptyEventLoop = false;
+
+  const token = event.headers.Authorization;
+  const decoded = decodeJwt(token);
+  if (!decoded) // token empty or invalid
+    return { 
+      statusCode: 403,
+      body: JSON.stringify({ message: 'Forbidden: missing or invalid JWT.'  })
+    };
+
+  return await sendRequest(decoded.id, JSON.parse(event.body))
+    .then(resp => ({
+      statusCode: 200,
+      body: JSON.stringify(resp)
+    }))
+    .catch(err => ({
+      statusCode: err.statusCode || 500,
+      body: JSON.stringify({ stack: err.stack, message: err.message })
+    }));
+};
+
+/*
+ * POST /friends/request/accept/
+ */
+module.exports.acceptRequest = async (event, context) => {
+  context.callbackWaitsForEmptyEventLoop = false;
+
+  const token = event.headers.Authorization;
+  const decoded = decodeJwt(token);
+  if (!decoded) // token empty or invalid
+    return { 
+      statusCode: 403,
+      body: JSON.stringify({ message: 'Forbidden: missing or invalid JWT.'  })
+    };
+
+  return await acceptRequest(decoded.id, JSON.parse(event.body))
+    .then(resp => ({
+      statusCode: 200,
+      body: JSON.stringify(resp)
+    }))
+    .catch(err => ({
+      statusCode: err.statusCode || 500,
+      body: JSON.stringify({ stack: err.stack, message: err.message })
+    }));
+};
+
+/*
+ * POST /friends/request/decline/
+ */
+module.exports.declineRequest = async (event, context) => {
+  context.callbackWaitsForEmptyEventLoop = false;
+
+  const token = event.headers.Authorization;
+  const decoded = decodeJwt(token);
+  if (!decoded) // token empty or invalid
+  return { 
+    statusCode: 403,
+    body: JSON.stringify({ message: 'Forbidden: missing or invalid JWT.'  })
+  };
+
+  return await declineRequest(decoded.id, JSON.parse(event.body))
+    .then(resp => ({
+      statusCode: 200,
+      body: JSON.stringify(resp)
+    }))
+    .catch(err => ({
+      statusCode: err.statusCode || 500,
+      body: JSON.stringify({ stack: err.stack, message: err.message })
+    }));
+};
+
+/**
+ * Helpers
+ */
+
+function decodeJwt(token) {
```

Comment:
Why is this method in the friends handler? This should be reused across the whole backend.